examples/complex.ml

```diff
@@ -0,0 +1,15 @@
+open Joy.Shape 
+
+(* 
+  Complex shapes can also be created from lists of shapes.
+  This allows us to apply any of the libraries' transformations to a group of 
+  shapes, liek we're doing here with scale
+*)
+
+let rec range a b = if a > b then [] else a :: range (a + 1) b
+
+let () = 
+  init ();
+  let comp = List.map (fun i -> circle ~x:(int_of_float ((float_of_int i /. 4.0) *. 100.0)) ~y: 0 50) (range (-5) 5) in 
+  render_shape (scale 0.5 (complex comp));
```

Comment:
Can we pease use a simpler example for illustration?

It'd be awesome if you come up with something creative for this. Maybe like applying some transformation on a group of shapes. Otherwise, take a donut and use translate on 2/3/4 instances of it to make an interesting pattern.

scale/rotate might change so better avoiding them from examples for the moment.

Reply:
I feel like I can either manage creative or simple but both will be a bit difficult.

My first thought, moving towards 'creative', was something like this:
```ocaml

(* creates a list containing numbers between a and b *)
let rec range a b = if a > b then [] else a :: range (a + 1) b

(* creates a list of (int * int) tuples,
   containing every combination of the elements of the two passsed lists *)
let cartesian_product l l' =
  List.concat (List.map (fun e -> List.map (fun e' -> (e, e')) l') l)

let () =
  init ();
  (* radius which also acts as grid spacing *)
  let radius = 50 in
  (* creating a grid with cartesian_product *)
  let coords = cartesian_product (range (-5) 5) (range (-5) 5) in
  (* using map to turn that into a complex shape that is a grid of circles *)
  let complex_shape =
    complex
      (List.map
         (fun (x, y) -> circle ~x:(x * radius) ~y:(y * radius) radius)
         coords)
  in
  (* translating that complex shape by radius / 2 *)
  let complex_transformed = translate (radius / 2) (radius / 2) complex_shape in
  show [ complex_shape; complex_transformed ];
  close ()
``` 

Which results in this lattice like pattern:
<img width="500" alt="Screen Shot 2023-10-21 at 10 50 20 PM" src="https://github.com/Sudha247/ocaml-joy/assets/95594152/0d9eac0b-77c5-4199-b831-00bea4d06744">
<img width="500" alt="Screen Shot 2023-10-21 at 10 54 23 PM" src="https://github.com/Sudha247/ocaml-joy/assets/95594152/47ce2150-ebd4-49aa-a289-ad46ad2c35a4">

I can certainly sit down and aim more for simplicity in the next couple of days though, if that'd be preferred!

Reply:
This is really beautiful, thanks!

> I feel like I can either manage creative or simple

By 'creative' here, I meant something that isn't too trivial only :p. I should've been clearer about that, sorry.

Reply:
Can you please push this  example? It looks good to me and we can go ahead with the merge.

Reply:
Done :)